tutorials/fit/NumericalMinimization.py

```diff
@@ -0,0 +1,85 @@
+#  \file
+#  \ingroup tutorial_fit
+#  \notebook -nodraw
+#  Example on how to use the new Minimizer class in ROOT
+#   Show usage with all the possible minimizers.
+#  Minimize the Rosenbrock function (a 2D -function)
+#  This example is described also in
+#  http://root.cern.ch/drupal/content/numerical-minimization#multidim_minim
+#  input : minimizer name + algorithm name
+#  randomSeed: = <0 : fixed value: 0 random with seed 0; >0 random with given seed
+#
+#  \macro_code
+#
+#  \author Lorenzo Moneta
+
+import ROOT
+
+def RosenBrock( vecx ):
+   x = vecx[0]
+   y = vecx[1]
+   return (y-x**2)**2 + (1-x)**2
+
+# create minimizer giving a name and a name (optionally) for the specific algorithm
+#  possible choices are:
+#      minimizerName                  algoName
+#     Minuit                     Migrad, Simplex,Combined,Scan  (default is Migrad)
+#      Minuit2                    Migrad, BFGS, Simplex,Combined,Scan  (default is Migrad)
+#     GSLMultiMin                ConjugateFR, ConjugatePR, BFGS, BFGS2, SteepestDescent
+#     GSLSimAn
+#     Genetic
+def NumericalMinimization(minimizerName = "Minuit2",
+                          algoName = "" ,
+                          randomSeed = -1) :
+
+   minimizer = ROOT.Math.Factory.CreateMinimizer(minimizerName, algoName);
+   if (not minimizer) :
+      print("Error: cannot create minimizer \"",minName,"\". Maybe the required library was not built?")
+      return 1;
+
+   # set tolerance and other minimizer parameters
+   # one can also use default values
+   minimizer.SetMaxFunctionCalls(1000000); # working for Minuit/Minuit2
+   minimizer.SetMaxIterations(10000);  # for GSL minimizers - no effect in Minuit/Minuit2
+   minimizer.SetTolerance(0.001);
+   minimizer.SetPrintLevel(1);
+
+   # create function wrapper for minimizer
+
+   f = ROOT.Math.Functor(RosenBrock,2);
+
+   #evaluate function at a point
+   import numpy as np
+   x0 = np.array([-1.,2.])
+   print("f(-1,1.2) = ",f(x0))
+
+   # starting point
+   variable = [-1.,1.2]
+   step = [0.01,0.01]
+   if (randomSeed >= 0) :
+      r = ROOT.TRandom2(randomSeed)
+      variable[0] = r.Uniform(-20,20);
+      variable[1] = r.Uniform(-20,20);
+
+   minimizer.SetFunction(f);
+
+   # Set the free variables to be minimized !
+   minimizer.SetVariable(0,"x",variable[0], step[0]);
+   minimizer.SetVariable(1,"y",variable[1], step[1]);
+
+   # do the minimization
+   ret = minimizer.Minimize();
+
+   xs = minimizer.X();
+   print("Minimum: f(", xs[0], ",", xs[1],"): ",minimizer.MinValue())
+
+   if ( ret and minimizer.MinValue()  < 1.E-4  and f(xs) < 1.E-4) :
+      print("Minimizer " , minimizerName," - ",algoName,
+             "converged to the right minimum!")
+   else :
+      print("Minimizer " ,minimizerName, " - ",algoName, "failed to converge !!!");
+      ROOT.Error("NumericalMinimization","fail to converge!");
+
+   return 0;
+
+NumericalMinimization()
```

Comment:
Calling the function uncoditionally at script level is not good practice. It would be better to use the syntax

```python
if __name__ == "__main__":
    raise SystemExit(NumericalMinimization())
```
The use of `SystemExit` makes the return values of the `NumericalMinimization` function actually useful. An alternative would be using the simpler notation:
```python
if __name__ == "__main__":
    NumericalMinimization()
```
But in this case, there should be no return values from the function. Up to you what you choose